packages/protocol/src/session/SessionManager.ts

```diff
@@ -533,13 +533,14 @@ export class SessionManager {
 
         this.#observers.close();
         await this.#storeResumptionRecords();
-        for (const session of this.#sessions) {
+        const closePromises = this.#sessions.map(async session => {
             await session?.end(false);
             this.#sessions.delete(session);
-        }
+        });
         for (const session of this.#insecureSessions.values()) {
-            await session?.end();
+            closePromises.push(session?.end());
         }
+        await Promise.allSettled(closePromises);
```

Comment:
Errors on close seem less problematic but would probably be worth handling here too